ablySpec/RestClient.swift

```diff
@@ -45,6 +45,32 @@ class RestClient: QuickSpec {
                     expect(publishTask.error?.code).toEventually(equal(40005))
                 }
 
+                it("should accept a token") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRest(key: options.key!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.auth.requestToken(ARTAuthTokenParams(), withOptions: nil) { tokenDetails, error in
+                            if let e = error {
+                                XCTFail(e.description)
+                                done()
+                                return
+                            }
+
+                            guard let tokenDetails = tokenDetails else {
+                                XCTFail("expected tokenDetails not to be nil when error is nil")
+                                done()
+                                return
+                            }
+
+                            let client = ARTRest(token: tokenDetails.token)
```

Comment:
It can be done, but it's strongly anti-idiomatic. All constructors that take arguments are like `initWithSomething:something`. So we have `initWithOptions`, `initWithToken`, `initWithKey`.